### ISSUE TITLE
Steamoverlay not working with dxvk enabled games

### ISSUE THREAD

**Issue by Cloudef** [ValveSoftware/Proton#1743]
I'm not sure if this is issue with my specific setup, but it seems steamoverlay doesn't work with any dxvk games. On the otherhand steamoverlay works with wine+dxvk when using khronos vulkan loader. I don't see any information regarding overlay compatibility and requirements on proton readme, nor any idea how to debug this issue.

Does proton use the native overlay? If so, does it need some prerequisites for vulkan enabled games? I have no native linux vulkan games.

**Comment by ananace** [comment]
Try launching the game with `VK_LOADER_DEBUG=all`, that'll output information about all the actions taken by the vulkan loader.

I had a similar issue due to the steamoverlay layers ending up getting malplaced, which is probably rather situational to me due to the messy setup. But you can check in `~/.local/share/vulkan/implicit_layers.d` and make sure there are steam overlay layers in there, and also check that the path is searched by the loader.  
If the loader for some reason doesn't search that path, you can always copy the layers to one of the manifest paths it reports searching through.

**Comment by Cloudef** [comment]
@ananace Thank you very much. This is exactly the kind of information I was looking for. I'll post my results later.

**Comment by Cloudef** [comment]
Ok it seems, whatever writes the implicit layers, doesn't honor XDG user dirs spec. So the layers ended up in `~/.local/share/vulkan/implicit_layer.d` instead of `${XDG_DATA_HOME:-~/.local/share}/vulkan/implicit_layer.d` Moving them to the correct place fixes this. However there seems to be issue with the steamoverlay with mouse cursor not being visible.

**Comment by kisak-valve** [comment]
Hello @Cloudef, the mouse cursor issue is being tracked at #168.

**Comment by Cloudef** [comment]
@kisak-valve Thank you. I'll track that. This issue will be fixed by honoring XDG user dirs when writing the layers.

**Comment by user9957** [comment]
According to the XDG spec (https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) :
> If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

So, `${XDG_DATA_HOME:-~/.local}/vulkan/implicit_layer.d` is not the correct place, and the program/lib expecting this is wrong.

**Comment by Cloudef** [comment]
Sorry, I meant `${XDG_DATA_HOME:-~/.local/share}/vulkan/implicit_layer.d`

**Comment by fls2018** [comment]
I've noticed myself using DXVK hud with some games causes the steam overlay not to load.

**Comment by user9957** [comment]
@Cloudef I see. I do have the files there, but the overlay doesn't work.

**Comment by Cloudef** [comment]
@user9957 Your issue may not be related to this issue. However try the suggestions made by @ananace to see if your issue is related to the overlay not loading or something else. Verify if your overlay works in other games (non-vulkan for example, since for opengl the overlay works very differently). Also check steam logs if it failed to LD_PRELOAD the overlay shared library.

**Comment by user9957** [comment]
You're right. I tested with an older game (not using d3d11) and the overlay doesn't show up as well. This is a steam play issue though, because the overlay works fine in native games.

For d3d11 games (using dxvk and vulkan), `VK_LOADER_DEBUG=all` doesn't add any output. I also tried with the vulkan cube demo, but setting the environment variable has no effect.

**Comment by momumi** [comment]
Had the same issue as @Cloudef, making a symlink fixed it:  

```sh
ln -s $XDG_DATA_HOME $HOME/.local/share
```

It looks like [vulkan-loader](https://github.com/KhronosGroup/Vulkan-Loader/commit/6e3ebec2141f39a7f850c8b4b966c176dac87c0f) handles this correctly, but steam must create the files in `$HOME/.local/share/vukan/`  

[Related issue](https://github.com/ValveSoftware/steam-for-linux/issues/1890).
